src/natcap/invest/ndr/ndr.py

```diff
@@ -834,11 +847,15 @@ def validate(args, limit_to=None):
                 (['calc_n', 'calc_p'],
                  'Either calc_n or calc_p must be True'))
 
-        LOGGER.debug('Required keys in CSV: %s', required_fields)
+        LOGGER.debug('Required nutrient-specific keys in CSV: %s',
+                     nutrient_required_fields)
+        # Check that these nutrient-specific keys are in the table
+        # validate has already checked all the other keys
         error_msg = validation.check_csv(
-            args['biophysical_table_path'], required_fields=required_fields)
-        LOGGER.debug('Error: %s', error_msg)
+            args['biophysical_table_path'],
+            header_patterns=nutrient_required_fields)
```

Comment:
sure, the `load_x`, `eff_x`, `crit_len_x` columns are included in the args spec (1) to document them and (2) to validate that they exist at all (at least one column matches each pattern). `validate` can't validate, and the args spec can't store, the more complicated relationship between these columns and the `calc_n`/`calc_p` inputs. It's not checking that when `calc_n` is `True`, the letter at the end of the column names must be `n` and not `p`. So that's checked in the model's custom validate function instead.